lib/resourcebuilder/batch.go

```diff
@@ -46,15 +46,25 @@ func checkJobHealth(ctx context.Context, client batchclientv1.JobsGetter, job *b
 	}
 
 	// Since we have filled in "activeDeadlineSeconds",
-	// the Job will 'Active == 0' if and only if it exceeds the deadline.
+	// the Job will 'Active == 0' if and only if it exceeds the deadline or if the update image could not be pulled.
 	// Failed jobs will be recreated in the next run.
-	if j.Status.Active == 0 && j.Status.Failed > 0 {
-		reason := "DeadlineExceeded"
-		message := "Job was active longer than specified deadline"
-		if len(j.Status.Conditions) > 0 {
-			reason, message = j.Status.Conditions[0].Reason, j.Status.Conditions[0].Message
+	if j.Status.Active == 0 {
+		// If there is more than one failed job pod then get the cause for failure
+		if j.Status.Failed > 0 {
+			reason := "DeadlineExceeded"
+			message := "Job was active longer than specified deadline"
+			if len(j.Status.Conditions) > 0 {
+				reason, message = j.Status.Conditions[0].Reason, j.Status.Conditions[0].Message
+			}
+			return false, fmt.Errorf("deadline exceeded, reason: %q, message: %q", reason, message)
+		}
+
+		// When the update image cannot be pulled then the pod is not marked as Failed, but the status condition is set
+		// after the job deadline is exceeded.
+		if len(j.Status.Conditions) > 0 && j.Status.Conditions[0].Reason == "DeadlineExceeded" {
+			return false, fmt.Errorf("deadline exceeded, reason: %q, message: %q", j.Status.Conditions[0].Reason, j.Status.Conditions[0].Message)
 		}
-		return false, fmt.Errorf("deadline exceeded, reason: %q, message: %q", reason, message)
```

Comment:
Lets add "unknown error" to the case where Active=0 and no Conditions set (that probably means some serious kube-api/scheduler issue)

Reply:
That condition will be true for a short period of time between the job being created and the first pod being created. So it might give false positives.

Reply:
Lets add some logging to this event, seems several occurrences might indicate a problem with the platform - I'd rather have CVO find it. Setting status for false positives is a bad idea indeed, but logging won't hurt

Reply:
Added the log